lib/gprocess.c

```diff
@@ -316,28 +354,33 @@ g_process_cap_restore(cap_t r)
 gboolean
 g_process_check_cap_syslog(void)
 {
-  int ret;
 
   if (have_capsyslog)
     return TRUE;
 
-  if (CAP_SYSLOG == -1)
-    return FALSE;
+  switch (_check_and_get_cap_from_text("cap_syslog", &cap_syslog))
+    {
+    case CAP_NOT_SUPPORTED_BY_LIBCAP:
+      fprintf (stderr, "The CAP_SYSLOG is not supported by libcap;"
+               "Falling back to CAP_SYS_ADMIN!\n");
+      return FALSE;
+      break;
 
-  ret = prctl(PR_CAPBSET_READ, CAP_SYSLOG);
-  if (ret == -1)
-    return FALSE;
+    case CAP_NOT_SUPPORTED_BY_KERNEL:
+      fprintf (stderr, "CAP_SYSLOG seems to be supported by libcap, but "
+               "the kernel can't parse it. Falling back to CAP_SYS_ADMIN!\n");
```

Comment:
A minor technicality, but the kernel does not do any parsing, so I'd recommend a slightly different wording:

```c
fprintf(stderr, "CAP_SYSLOG seems to be supported by libcap, but "
        "the kernel does not appear to recognize it. Falling back "
        "to CAP_SYS_ADMIN!\n");
```